packages/plugins/plugin-table/src/model/table-model.ts

```diff
@@ -398,6 +409,15 @@ export class TableModel<T extends BaseTableRow = {}> extends Resource {
     }
   }
 
+  //
+  // Interactions
+  //
+  public handleGridClick = (event: React.MouseEvent): void => {
```

Comment:
i don't think this is right. the model shouldn't be directly involved in the UX. (in principle the model could be used by a terminal app.) otherwise it'll blur the line of responsibility. i think the table should create and manage the ModalController separately, just like it does to manage the toolbar. splitting this also reduces the exposure of model to other components like RowActionMenu.

(also, all `React` defs should be imported in the `react` import above.)

Reply:
I'm going to push back on this.

1. Table model is already tied intimately with grid and constructing grid cells.
2. Table model currently wires up and renders the accessory html for each modal trigger button including setting the attributes.
3. Table model is about to become even more integrated with the UX, when we start rendering checkboxes for selection, toggles for Boolean cells, better triggers for refs at best, and tag lists in multi-select cells. We need a standard way of managing interactions and reacting to each of these being clicked since they have stateful changes and I think it should live in the model, unless we workout a way to abstract the entire presentational layer (which I think we should do all at once if we want that).

Reply:
fwiw one of the reasons I made it a separate class is so that presentation can get abstracted later.